controllers/util/util.go

```diff
@@ -96,3 +99,9 @@ func CalculateMD5Hash(value any) string {
 	hash := md5.Sum(data)
 	return hex.EncodeToString(hash[:])
 }
+
+func AssertEqual(actual any, expected any, exitCode int) {
+	if !reflect.DeepEqual(actual, expected) {
+		os.Exit(exitCode)
+	}
+}
```

Comment:
There is no reason or need to use reflect in places where reflection is not needed. Reflection is heavy and can introduce multiple runtime issues. I don't see a reason to invoke reflection where the current use case is equality between simple values. And in case we need the deep equal (the case of nested structures that holds pointers) we should reflect that in the name of the func (call it `AssertDeepEqual`)

```suggestion
func AssertEqual[T comparable](actual T, expected T, exitCode int) {
	if actual != expected {
		os.Exit(exitCode)
	}
}
```